agimus_controller/agimus_controller/mpc.py

```diff
@@ -0,0 +1,58 @@
+import time
+from agimus_controller.ocp_base import OCPBase
+from agimus_controller.warm_start_base import WarmStartBase
+from agimus_controller.trajectory import (
+    TrajectoryPoint,
+    WeightedTrajectoryPoint,
+    TrajectoryBuffer,
+)
+from agimus_controller.mpc_data import OCPResults, MPCDebugData
+
+
+class MPC(object):
+    def __init__(self) -> None:
+        self._ocp = None
+        self._warm_start = None
+        self._mpc_debug_data: MPCDebugData = None
+        self._buffer = TrajectoryBuffer()
+
+    def setup(self, ocp: OCPBase, warm_start: WarmStartBase) -> None:
+        self._ocp = ocp
+        self._warm_start = warm_start
```

Comment:
```suggestion

    def setup(self, ocp: OCPBase, warm_start: WarmStartBase, buffer: TrajectoryBuffer = TrajectoryBuffer()) -> None:
        self._ocp = ocp
        self._warm_start = warm_start
        self._buffer = buffer
```
Maybe implementation like this? It assumes the default buffer implementation is used, while leaving some flexibility to replace it for cases with variable `dt` and so on